src/rust/rust-proto-new/tests/integration_test.rs

```diff
@@ -235,18 +259,174 @@ proptest! {
     //
 
     #[test]
-    fn test_publish_raw_logs_request_encode_decode(
-        publish_raw_logs_request in publish_raw_logs_requests()
+    fn test_publish_raw_log_request_encode_decode(
+        publish_raw_log_request in publish_raw_log_requests()
     ) {
-        check_encode_decode_invariant(publish_raw_logs_request)
+        check_encode_decode_invariant(publish_raw_log_request)
     }
 
     #[test]
-    fn test_publish_raw_logs_response_encode_decode(
-        publish_raw_logs_response in publish_raw_logs_responses()
+    fn test_publish_raw_log_response_encode_decode(
+        publish_raw_log_response in publish_raw_log_responses()
     ) {
-        check_encode_decode_invariant(publish_raw_logs_response)
+        check_encode_decode_invariant(publish_raw_log_response)
     }
 
     // TODO: add more here as they're implemented
 }
+
+//
+// ---------------- gRPC tests -------------------------------------------------
+//
+// These tests exercise the gRPC machinery. The idea here is to exercise an
+// API's success and error paths with a simple mocked business logic
+// implementation. The gRPC client should be used to exercise the gRPC
+// server.
+//
+
+// We can use this global TENANT_ID for two purposes:
+//   (1) To make sure some test data made it through to our mocked
+//       application layer.
+//   (2) To send a poison pill to our mocked application layer, to
+//       trigger an error response.
+const TENANT_ID: &'static str = "f000b11e-b421-4ffe-87c2-a963b77fd8e9";
```

Comment:
I think these hardcoded IDs should be sufficiently random to avoid collisions even in the case where they're subject to things like shrinking in `proptest` tests. For example, if I made this ID all `0`s or all `f`s I might expect a proptest or similar strategy to hit those values while shrinking.

While this collision wouldn't be catastrophic for the `TENANT_ID`, it would cause a false-positive with the `BAD_EVENT_SOURCE_ID`.